pkg/storage/unified/resource/client.go

```diff
@@ -142,3 +142,86 @@ func allowInsecureTransportOpt(grpcClientConfig *authnlib.GrpcClientConfig, opts
 	tokenClient, _ := authnlib.NewTokenExchangeClient(*grpcClientConfig.TokenClientConfig, authnlib.WithHTTPClient(client))
 	return append(opts, authnlib.WithTokenClientOption(tokenClient))
 }
+
+var (
+	_ ResourceClient = (*directResourceClient)(nil)
+)
+
+// The direct client passes requests directly to the server using the *same* context
+func NewDirectResourceClient(server ResourceServer) ResourceClient {
+	return &directResourceClient{server}
+}
```

Comment:
Note that if we decide to enable auth in local mode, this will require setting it up separately.

Reply:
right -- for the "direct" mode authentication already exists in context